src/traces/indicator/defaults.js

```diff
@@ -0,0 +1,162 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+var attributes = require('./attributes');
+var handleDomainDefaults = require('../../plots/domain').defaults;
+var Template = require('../../plot_api/plot_template');
+var handleArrayContainerDefaults = require('../../plots/array_container_defaults');
+var cn = require('./constants.js');
+
+var handleTickValueDefaults = require('../../plots/cartesian/tick_value_defaults');
+var handleTickMarkDefaults = require('../../plots/cartesian/tick_mark_defaults');
+var handleTickLabelDefaults = require('../../plots/cartesian/tick_label_defaults');
+
+function supplyDefaults(traceIn, traceOut, defaultColor, layout) {
+    function coerce(attr, dflt) {
+        return Lib.coerce(traceIn, traceOut, attributes, attr, dflt);
+    }
+
+    handleDomainDefaults(traceOut, layout, coerce);
+
+    // Mode
+    coerce('mode');
+    traceOut._hasNumber = traceOut.mode.indexOf('number') !== -1;
+    traceOut._hasDelta = traceOut.mode.indexOf('delta') !== -1;
+    traceOut._hasGauge = traceOut.mode.indexOf('gauge') !== -1;
+
+    coerce('value');
+    coerce('vmin');
+    coerce('vmax', 1.5 * traceOut.value);
+
+    // Number attributes
+    var auto = [];
+    var bignumberFontSize;
+    if(traceOut._hasNumber) {
+        coerce('number.valueformat');
+        coerce('number.font.color', layout.font.color);
+        coerce('number.font.family', layout.font.family);
+        coerce('number.font.size', 'auto');
+        if(traceOut.number.font.size === 'auto') {
+            traceOut.number.font.size = cn.defaultNumberFontSize;
+            auto[0] = true;
+        }
+        coerce('number.suffix');
+        bignumberFontSize = traceOut.number.font.size;
+    }
+
+    // delta attributes
+    var deltaFontSize;
+    if(traceOut._hasDelta) {
+        coerce('delta.font.color', layout.font.color);
+        coerce('delta.font.family', layout.font.family);
+        coerce('delta.font.size', 'auto');
+        if(traceOut.delta.font.size === 'auto') {
```

Comment:
We probably don't need to handle a special `'auto'` value. Omitting the `dflt` in `*.font.size` attribute declarations, we can handle the case where e.g. here `traceOut.delta.font.size` is `undefined`.

Reply:
Done in e6d8525ab07c56ca10bdbac6762b06d1d57bf830

Reply:
Marking as resolved. Follow-up in https://github.com/plotly/plotly.js/pull/3978#pullrequestreview-254682391